src/kernels.jl

```diff
@@ -0,0 +1,411 @@
+
+abstract type KNNKernel end
+
+sort_idxs(::KNNKernel) = false
+
+# Function that lists all defined `KNNKernels`
+list_kernels() = subtypes(KNNKernel) 
+
+####
+# User Defined Kernel (UDK)
+####
+
+_nothing(x) = nothing
+
+"""
+    UDK(;func::Function = x->nothing, sort::Bool=false)
+
+Wrap a user defined nearest neighbors weighting function `func` as a `KNNKernel`.
+ 
+# Keywords
+
+- `func` : user-defined nearest neighbors weighting function. The function 
+   should have the signature `func(dists_matrix)::Union{Nothing, <:AbstractMatrix}`, 
+   where `dists_matrix` is the k-nearest neighbors distances matrix and outputs a matrix 
+   of the same shape as `dists_matrix`.   
```

Comment:
Add something like "A distance matrix has shape `(p, k)` where `p` is the dimension of the feature space, and `k` the number of observations to be weighted".

Reply:
I also suggest `UserDefinedKernel` instead of `UDK`. If this leads to a lot of long lines, you could create an alias.